src/autogluon/bench/datasets/multimodal_dataset.py

```diff
@@ -265,8 +302,12 @@ class SNLI(BaseMatcherDataset):
     def __init__(self, split="train"):
         self._split = split
         self._path = os.path.join(get_data_home_dir(), "snli", f"{split}.csv")
-        download(self._INFO[split]["url"], path=self._path, sha1_hash=self._INFO[split]["sha1sum"])
-        self._data = pd.read_csv(self._path, delimiter="|")
+        try:
```

Comment:
This code block seems to be repeated many times in different classes. Shall we put them into the base class instead to avoid duplication>

Reply:
Good point. I have refactored the code.